src/braket/default_simulator/simulator.py

```diff
@@ -393,10 +400,15 @@ def _formatted_measurements(simulation: Simulation) -> list[list[str]]:
             list[list[str]]: List containing the measurements, where each measurement consists
             of a list of measured values of qubits.
         """
-        return [
+        # Get the full measurements
+        measurements = [
             list("{number:0{width}b}".format(number=sample, width=simulation.qubit_count))
             for sample in simulation.retrieve_samples()
         ]
+        #  Gets the subset or measurements from the full measurements
+        if measured_qubits is not None and measured_qubits != []:
+            measurements = [[row[i] for i in measured_qubits] for row in measurements]
```

Comment:
I'm not sure that this would work since `sample` would be an integer so it cannot be indexed. 

For example, `simulation.retrieve_samples()` return something like:
`[7 0 7 1 7 0  1 0  0  1]`

and after the formatting returns the measurement list as 0s and 1s:
`[['1', '1', '0'], 
   ['1', '1', '1'], 
   ['0', '0', '1'], 
   ['0', '0', '1'], 
   ['1', '1', '0'], 
   ['1', '1', '0'], 
   ['1', '1', '1'], 
   ['1', '1', '1'], 
   ['0', '0', '1'], 
   ['1', '1', '1']]`

Reply:
Oh fair enough. We could probably still simplify the nested list comprehension by using np indexing, but it's not the most important

Reply:
Updated to use np indexing like so: `measurements = np.array(measurements)[:, measured_qubits].tolist()`